src/ducks/categorization/services.js

```diff
@@ -213,24 +216,54 @@ export const localModel = async (classifierOptions, transactions) => {
       'No classifier, impossible to categorize transactions'
     )
     return
+  } else {
+    return classifier
   }
+}
+
+export const pctOfTokensInVoc = (tokens, vocabularyArray) => {
+  const n_tokens = tokens.length
+  const intersection = tokens.filter(t => -1 !== vocabularyArray.indexOf(t))
+  return intersection.length / n_tokens
+}
+
+export const localModel = async (classifierOptions, transactions) => {
+  const classifier = await createLocalModel(classifierOptions)
 
   localModelLog('info', `Applying model to ${transactions.length} transactions`)
+  const vocabulary = Object.keys(classifier.vocabulary)
   for (const transaction of transactions) {
     const label = getLabelWithTags(transaction)
-    localModelLog('info', `Applying model to ${label}`)
-
-    const { category, proba } = maxBy(
-      classifier.categorize(label).likelihoods,
-      'proba'
-    )
-
-    transaction.localCategoryId = category
-    transaction.localCategoryProba = proba
-
-    localModelLog('info', `Results for ${label} :`)
-    localModelLog('info', `localCategory: ${category}`)
-    localModelLog('info', `localProba: ${proba}`)
+    // First : check if tokens from the transaction's label are in the model
+    const tokens = tokenizer(transaction.label)
+    const pctOfThisTokensInVoc = pctOfTokensInVoc(tokens, vocabulary)
+    if (pctOfThisTokensInVoc > LOCAL_MODEL_PCT_TOKENS_IN_VOC_THRESHOLD) {
+      // If OK : continue. Otherwise : abort with category '0' and proba 1/nbUniqueCat
+      localModelLog('info', `Applying model to ${label}`)
+  
+      const { category, proba } = maxBy(
+        classifier.categorize(label).likelihoods,
+        'proba'
+      )
+  
+      transaction.localCategoryId = category
+      transaction.localCategoryProba = proba
+  
+      localModelLog('info', `Results for ${label} :`)
+      localModelLog('info', `localCategory: ${category}`)
+      localModelLog('info', `localProba: ${proba}`)
+    } else {
+      // If KO : abort with category '0' and proba 1/nbUniqueCat
+      localModelLog('info', `Giving up for ${label}`)
+      const category = LOCAL_MODEL_CATEG_FALLBACK
+      const proba = LOCAL_MODEL_PROBA_FALLBACK
+      transaction.localCategoryId = category
+      transaction.localCategoryProba = proba
+  
+      localModelLog('info', `Results for ${label} :`)
+      localModelLog('info', `localCategory: ${category}`)
+      localModelLog('info', `localProba: ${proba}`)
```

Comment:
These 3 lines are duplicated in both `if/else` branches. You can put it outside the conditions.